File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:

        n = len(nums)

        dp = [0] * n
        dp[n - 1] = 1

        for i in range(n - 2, -1, -1):

            for jump in range(1, nums[i] + 1):

                if i + jump < n and dp[i + jump] == 1:
                    dp[i] = 1
                    break

        return dp[0] == 1

File: Data Structures & Algorithms/majority-element-ii/submission-0.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        ArrayList<Integer> result = new ArrayList<>();

        for (int num : nums) {
            if (!map.containsKey(num)) {
                map.put(num, 1);
            } else {
                map.put(num, map.get(num) + 1);
            }
        }

        for (int key : map.keySet()) {
            if (map.get(key) > nums.length / 3) {
                result.add(key);
            }
        }

        return result;
        
    }
}